File: TypeScript/769.最多能完成排序的块.ts
function maxChunksToSorted(arr: number[]): number {
    let ans: number = 0
    for (let [i, maxn] = [0, Number.MIN_SAFE_INTEGER]; i < arr.length; ++i) {
        maxn = Math.max(maxn, arr[i])
        if (i == maxn)
            ++ans
    }
    return ans
};

File: 769.最多能完成排序的块.cc
#include <vector>
#include <algorithm>
using std::vector;
using std::max;

class Solution {
public:
    int maxChunksToSorted(vector<int> &arr) {
        int ans = 0;
        for (int i = 0, maxn = 0; i < arr.size(); ++i) {
            maxn = max(maxn, arr[i]);
            if (maxn == i)
                ++ans;
        }
        return ans;
    }
};
